src/main/java/MyArrayList.java

```diff
@@ -0,0 +1,335 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.function.UnaryOperator;
+
+public class MyArrayList<T> implements List<T> {
+
+  private T[] tempTable;
+  private int size;
+
+  public MyArrayList() {
+    tempTable = (T[]) new Object[0];
+    size = 0;
+  }
+
+  public MyArrayList(Collection<? extends T> c) {
+
+    tempTable = (T[]) c.toArray();
+    size = tempTable.length;
+  }
+
+  public int size() {
+    return size;
+  }
+
+  public boolean isEmpty() {
+    return size == 0;
+  }
+
+  public boolean contains(final Object o) {
```

Comment:
please avoid using single letter names for variables - please correct in. all methods. and. tests.